map_view/display-drawing-status/src/main/java/com/esri/samples/display_drawing_status/DisplayDrawingStatusSample.java

```diff
@@ -87,23 +85,13 @@ public void start(Stage stage) {
       Envelope envelope = new Envelope(bottomLeftPoint, topRightPoint);
       mapView.setViewpoint(new Viewpoint(envelope));
 
-      mapView.addDrawStatusChangedListener(e -> {
-        // check to see if draw status is in progress
-        if (e.getDrawStatus() == DrawStatus.IN_PROGRESS) {
-          // reset progress bar as in progress
-          progressBar.setProgress(-100.0);
-
-          // check to see if draw status is complete
-        } else if (e.getDrawStatus() == DrawStatus.COMPLETED) {
-          // set progress bar as complete
-          progressBar.setProgress(100.0);
-        }
-      });
-
-      // add the map view and progress bar to stack pane
-      stackPane.getChildren().addAll(mapView, progressBar);
-      StackPane.setAlignment(progressBar, Pos.TOP_LEFT);
-      StackPane.setMargin(progressBar, new Insets(10, 0, 0, 10));
+      // bind progress indicator visibility to map view draw status
+      progressIndicator.visibleProperty().bind(mapView.drawStatusProperty().isEqualTo(DrawStatus.IN_PROGRESS));
+
+      // add the map view and progress indicator to stack pane
+      stackPane.getChildren().addAll(mapView, progressIndicator);
+      StackPane.setAlignment(progressIndicator, Pos.TOP_LEFT);
+      StackPane.setMargin(progressIndicator, new Insets(10, 0, 0, 10));
```

Comment:
```suggestion
      StackPane.setAlignment(progressIndicator, Pos.CENTER);
```
As discussed, we should consider moving this to the center now it's a progress indicator rather than bar. And - sorry - the image would need to reflect that too!